src/pretalx/common/mixins/models.py

```diff
@@ -38,3 +39,24 @@ def logged_actions(self):
             content_type=ContentType.objects.get_for_model(type(self)),
             object_id=self.pk,
         ).select_related('event', 'person')
+
+
+class IdBasedSlug:
+    """
+    Adds a method to retrieve a human-understandable slug based on the `id` field of a model.
+    """
+
+    slug_separator = '-'
+
+    def slug(self):
+        """
+        Get slug of this object.
+        """
+        return f'{self.id}{self.slug_separator}{slugify(self.name)}'
+
+    @classmethod
+    def id_from_slug(cls, slug):
+        """
+        Get ID from slug value.
+        """
+        return int(slug.split(cls.slug_separator)[0])
```

Comment:
This potentially causes a `ValueError` if called with a string that doesn't start with a number.